src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO.Strategies
+{
+    internal static partial class FileStreamHelpers
+    {
+        // It's enabled by default. We are going to change that (by removing !) once we fix #16354, #25905 and #24847.
+        internal static bool UseLegacyStrategy { get; }
+            = !AppContextConfigHelper.GetBooleanConfig("System.IO.UseLegacyFileStream", "DOTNET_SYSTEM_IO_USELEGACYFILESTREAM");
```

Comment:
Let's not ship a preview in a state where someone setting UseLegacyFileStream actually gets the opposite behavior.

Reply:
@stephentoub @carlossanlop Do you have any name preferences?

* New 
   * `System.IO.UseLegacyFileStream` -> `System.IO.UseNewFileStream`
   * `DOTNET_SYSTEM_IO_USELEGACYFILESTREAM` -> `DOTNET_SYSTEM_IO_USENEWFILESTREAM`
* Experimental?
   * `System.IO.UseExperimentalFileStream` -> `System.IO.UseExperimentalFileStream`
   * `DOTNET_SYSTEM_IO_USEEXPERIMENTALFILESTREAM` -> `DOTNET_SYSTEM_IO_USEEXPERIMENTALFILESTREAM`

Reply:
or `DisableLegacyFileStream` ?

Reply:
I liked @jeffhandley's suggestion, which was something like UseNet5CompatFileStream.

To be clear, though, I was commenting on the fact that as you have this coded, if someone sets UseLegacyFileStream to false, the code is actually going to use the legacy implementation, and if they set UseLegacyFileStream to true, it won't.  It's backwards.

Reply:
I've changed the name to `System.IO.UseNet5CompatFileStream` and removed the confusing negation. PTAL